File: NeetCode150/arrays_and_hashing/1_contains_duplicate.cc
// Space: O(n)
// Time: O(n)
class Solution {
    public:
        bool containsDuplicate(vector<int>& nums) {
            unordered_set<int> set;
    
            for (auto num: nums) {
                if (set.contains(num)) return true;
                set.insert(num);
            }
    
            return false;
        }
    };

File: NeetcodeBlind75/arrays_and_hashing/1_contains_duplicate.py
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        """
        Algorithm intuition:
            - we can use a set
            - iteratively go through the all the elements
            - if any value is already in the set, we have found a duplicate and return true, else false
        Time: O(n)
        Space: O(n)
        """

        seen = set()

        for num in nums:
            if num in seen:
                return True
            seen.add(num)

        return False
